framework/src/nodalkernels/ADNodalKernel.C

```diff
@@ -54,7 +57,10 @@ ADNodalKernel::computeJacobian()
     const auto dof_idx = _var.nodalDofIndex();
     _qp = 0;
     const auto res = computeQpResidual();
-    _assembly.processJacobian(res, dof_idx, _matrix_tags);
+    addJacobian(_assembly,
```

Comment:
how many assembly are there? I m thinking this should just live in the tagging interface

Reply:
There are very rare instances where you don't want to use `_assembly` ... displaced nodal boundary conditions is one off the top of my head